core/runtime/TRTEngine.cpp

```diff
@@ -248,6 +245,32 @@ void TRTEngine::set_profiling_paths() {
   cuda_graph_debug_path = std::filesystem::path{profile_path_prefix + "/" + name + "_cudagraph.dot"}.string();
 }
 
+int64_t TRTEngine::get_weight_streaming_budget_v2() {
+  return cuda_engine->getWeightStreamingBudgetV2();
+}
+
+bool TRTEngine::set_weight_streaming_budget_v2(int64_t budget) {
+  return cuda_engine->setWeightStreamingBudgetV2(budget);
```

Comment:
If a new weight stream budget requires a new execution context, that should be done atomically with this API